airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/ReplicationView.tsx

```diff
@@ -68,14 +69,16 @@ export const ReplicationView: React.FC<ReplicationViewProps> = ({ onAfterSaveSch
 
   const onSubmit = async (values: ValuesProps, formikHelpers?: FormikHelpers<ValuesProps>) => {
     const initialSyncSchema = connection?.syncCatalog;
+    const connectionAsUpdate = connection ? toWebBackendConnectionUpdate(connection) : undefined;
 
     await updateConnection({
+      ...connectionAsUpdate,
```

Comment:
what happens here if `connectionAsUpdate` is undefined?

Reply:
There may be something deeper here but the connection should not be undefined if we're saving.

Reply:
I figured out what's going on. See the rest of this file. This `onSubmit` function is only called when there is a connction object, else it renders a loading indicator. I updated the code regardless to check if there is a connection object first.